### PR TITLE
Fix: ensure get_workflows_for_user returns only the user workflows

### PR DESCRIPTION
**Pull request by pieterlukasse**
Jira Ticket: [VADC-801](https://ctds-planx.atlassian.net/browse/VADC-801)

### Bug Fixes
- fix to ensure get_workflows_for_user returns only the user workflows meaning only workflows executed by the user OUTSIDE of a "team project"

[VADC-801]: https://ctds-planx.atlassian.net/browse/VADC-801?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/argowrapper/engine/argo_engine.py
Comment (tianj7) [inline src/argowrapper/engine/argo_engine.py:1]: I think the comment on 296 regarding return types also needs to be updated.

File: src/argowrapper/engine/argo_engine.py [hunk 1]
```diff
@@ -301,8 +303,12 @@ def _get_archived_workflow_given_name(self, archived_workflow_uid) -> str:
         workflow_details = self.get_workflow_details(None, archived_workflow_uid)
         #  get the workflow given name from the parsed details:
         given_name = workflow_details["wf_name"]
-        self.workflow_given_names_cache[archived_workflow_uid] = given_name
-        return given_name
+        team_project = workflow_details[GEN3_TEAM_PROJECT_METADATA_LABEL]
+        self.workflow_given_names_cache[archived_workflow_uid] = (
```
Comment (tianj7) [inline src/argowrapper/engine/argo_engine.py:307]: For my information, is the purpose of this line to set the value in cache for subsequent reads?

Comment (pieterlukasse) [reply src/argowrapper/engine/argo_engine.py:307]: correct!

File: src/argowrapper/engine/argo_engine.py
Comment (pieterlukasse) [reply src/argowrapper/engine/argo_engine.py:1]: good catch! Updated.

**Review by tianj7 — APPROVED** [review]
lgtm

**Pull request merged**